src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -74,6 +74,28 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
     datastore.put(currentRoom);
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String roomId = request.getParameter("roomId");
+
+    // Room query that looks for correct room.
```

Comment:
It looks like lines 82-85 in your doGet method are repeats of lines 63-67 in your doPut method. In both cases, you're using almost the same exact code to load a room by ID. In cases like these, it's best to extract the common code into a method that you can call in both places where the code is duplicated. E.g. the method signature could look like:
Entity loadRoom(String roomId);

This follows the "DRY Principle" (Don't Repeat Yourself), similar to my comment in lobby.js. Take a look at that comment for more info about the DRY Principle, and let me know if you have any questions about this.